src/operation/commit_transaction/mod.rs

```diff
@@ -33,7 +36,13 @@ impl Operation for CommitTransaction {
             Self::NAME: 1,
         };
 
-        append_options(&mut body, self.options.as_ref())?;
+        let mut options = self.options.clone().unwrap_or_default();
```

Comment:
It would be good to capture this pattern - either via a new trait implemented by the various `*Options` that gives access to a `&mut WriteConcern`, or a simple macro.

Reply:
Wrote a macro in https://github.com/mongodb/mongo-rust-driver/pull/513/commits/1f81c084161c53fafef01c2fff489a19bfda7ff2. `run_command`, `abort_transaction`, and `drop_database` go about this slightly differently, so I didn't use it in those three operations.